kartothek/io/dask/dataframe.py

```diff
@@ -73,9 +78,18 @@ def read_dataset_as_ddf(
         label_filter=label_filter,
         dates_as_object=dates_as_object,
         predicates=predicates,
+        dispatch_by=dask_index_on,
     )
-
-    return dd.from_delayed(delayed_partitions, meta=meta)
+    if dask_index_on:
+        divisions = ds_factory.indices[dask_index_on].observed_values()
+        divisions.sort()
+        divisions = list(divisions)
+        divisions.append(divisions[-1])
+        return dd.from_delayed(
+            delayed_partitions, meta=meta, divisions=divisions
```

Comment:
`divisions` shouldn't be specified here should it?

Reply:
Technically, at the moment, it doesn't do anything but I don't know if this will change

Reply:
please remove it

Reply:
@fjetter this is logicaly incorrect, no?

Reply:
no, it isn't, why would it be?

Reply:
Because at that point (i.e. before setting any index) the `divisions` would be referencing the default pandas integer/range index, not any custom index column. However, the variable `divisions` created references the divisions for the custom index column, which is not yet set on the dataframe (set in the next line, where it does make sense).

Reply:
Although the code runs fine, it shouldn't be there AFAI see